gui/layers/_vectors_layer/model.py

```diff
@@ -0,0 +1,624 @@
+#!/usr/bin/env python
+# title           : properties.py
+# description     :class Vectors layer that defines properties
+# author          :bryant.chhun
+# date            :1/16/19
+# version         :0.0
+# usage           :
+# notes           :
+# python_version  :3.6
+
+from typing import Union
+
+import numpy as np
+
+from .._base_layer import Layer
+from .._register import add_to_viewer
+from gui._vispy.scene.visuals import Line as LinesNode
+from vispy.color import get_color_names
+
+from vispy.util.event import Event
+
+from .view import QtVectorsLayer
+from .view import QtVectorsControls
+
+
+@add_to_viewer
+class Vectors(Layer):
+    """
+    Vectors layer renders lines onto the image.
+    There are currently two data modes supported:
+        1) image-like vectors where every pixel defines its own vector
+        2) coordinate-like vectors where x-y position is not fixed to a grid
+    Supports ONLY 2d vector field
+
+    Properties
+    ----------
+    vectors : np.ndarray
+        array of shape (N,4) or (N, M , 2)
+
+    data : np.ndarray
+        ABC implementation.
+        returns vectors property setter/getter
+
+    _original_data : np.ndarray
+        Used by averaging and length adjustments.
+        Is set during layer construction.
+        Is NOT updated if data is adjusted by assigning the 'vectors' property
+
+    averaging : str
+        kernel over which to average from one of _kernel_dict
+        averaging.setter adjusts the underlying data
+        subscribe an observer by registering it with "averaging_bind_to"
+
+    width : int
```

Comment:
As width increases, I encounter a point where the lines no longer increase in width (depending on my zoom). Do other's notice this too? I think this is a vispy (and possibly open-gl) problem under the hood, but just wanted to note here

Reply:
Interesting.  I am able to increase the width to about maximum 65 pixels in both demos.  Above that, the rendered image doesn't change..

Reply:
Hmm, mine still stops only around 3 pixels. It's a bug in the vispy stuff I think. I've seen some warnings about it in their docs. I have some alternative code for rendering lines, but I'm not sure we need to go down that path yet. Depends on how annoying people find it. Worth making a note in the docs maybe